ddl/table.go

```diff
@@ -773,7 +773,7 @@ func onModifyTableCharsetAndCollate(t *meta.Meta, job *model.Job) (ver int64, _
 
 func checkTableNotExists(d *ddlCtx, t *meta.Meta, schemaID int64, tableName string) error {
 	// d.infoHandle maybe nil in some test.
-	if d.infoHandle == nil {
+	if d.infoHandle == nil || d.infoHandle.IsNil() {
```

Comment:
Could we rename this function name? `IsNil` made me think that it is used to judge whether the value of `infoHandle` is empty.

Reply:
How about `Isvalid`

Reply:
I think `IsValid` is OK. But I see other one using `Valid` in our TiDB.